src/outagesim/backup_reliability.jl

```diff
@@ -396,106 +655,111 @@ function survival_with_battery(net_critical_load::Vector, starting_batt_soc_kwh:
 ```
 """
-function backup_reliability_inputs(d::Dict, p::REoptInputs; r::Dict = Dict())::Dict
+function backup_reliability_reopt_inputs(;d::Dict, p::REoptInputs, r::Dict = Dict())::Dict
     zero_array = zeros(length(p.time_steps))
+
+
     r2 = dictkeys_tosymbols(r)
-    critical_loads_kw = p.s.electric_load.critical_loads_kw
+    
+    net_critical_loads_kw = p.s.electric_load.critical_loads_kw
 
+    #TODO Change CHP to meet capacity not load
     if "CHP" in keys(d)
-        chp_generation =  get(d["CHP"], "year_one_electric_production_series_kw", zero_array)
-        critical_loads_kw .-= chp_generation
+        chp_generation =  get(d["CHP"], "size_kw", 0)
+        net_critical_loads_kw .-= chp_generation
     end
 
     microgrid_only = get(r, "microgrid_only", false)
 
     pv_kw_ac_hourly = zero_array
-    if "PV" in keys(d)
+    if !(
+            microgrid_only && 
+            !Bool(get(d, "PV_upgraded", false))
+        ) && 
+        "PV" in keys(d)
+
         pv_kw_ac_hourly = (
             get(d["PV"], "year_one_to_battery_series_kw", zero_array)
             + get(d["PV"], "year_one_curtailed_production_series_kw", zero_array)
             + get(d["PV"], "year_one_to_load_series_kw", zero_array)
             + get(d["PV"], "year_one_to_grid_series_kw", zero_array)
         )
     end
-    if microgrid_only && !Bool(get(d, "PV_upgraded", false))
-        pv_kw_ac_hourly = zero_array
-    end
 
-    batt_kwh = 0
-    batt_kw = 0
+    battery_size_kwh = 0
+    battery_size_kw = 0
     if "Storage" in keys(d)
         #TODO change to throw error if multiple storage types
         for b in p.s.storage.types.elec
-            batt_charge_efficiency = p.s.storage.attr[b].charge_efficiency
-            batt_discharge_efficiency = p.s.storage.attr[b].discharge_efficiency
+            battery_charge_efficiency = p.s.storage.attr[b].charge_efficiency
+            battery_discharge_efficiency = p.s.storage.attr[b].discharge_efficiency
         end
             
-        batt_kwh = get(d["Storage"], "size_kwh", 0)
-        batt_kw = get(d["Storage"], "size_kw", 0)
+        battery_size_kwh = get(d["Storage"], "size_kwh", 0)
+        battery_size_kw = get(d["Storage"], "size_kw", 0)
         init_soc = get(d["Storage"], "year_one_soc_series_pct", [])
 
         if microgrid_only && !Bool(get(d, "storage_upgraded", false))
-            batt_kwh = 0
-            batt_kw = 0
+            battery_size_kwh = 0
+            battery_size_kw = 0
             init_soc = []
         end
 
-        starting_batt_soc_kwh = init_soc .* batt_kwh
+        starting_battery_soc_kwh = init_soc .* battery_size_kwh
 
-        #Only adds PV generation if there is also a battery
-        critical_loads_kw .-= pv_kw_ac_hourly
-        r2[:starting_batt_soc_kwh] = starting_batt_soc_kwh
+        #Only subtracts PV generation if there is also a battery
```

Comment:
Should put this assumption in backup_reliability and backup_reliability_inputs function documentation (@sericson0) and API Django input help text (me)